### ISSUE TITLE
Do we need a specific syntax to list profile combinations?

### ISSUE THREAD

**Issue by larsgsvensson** [larsgsvensson/I-D-Accept--Schema#5]
I'm not yet quite sure which is the best way to structure the "Accept-Profile" header field. We have (at least) three different cases:
* The requested resource adheres to one and only one profile (e. g. an XML document must validate against one particular schema). This one is fairly straightforward since we only need specify one URI reference.
* The requested resource should use multiple profiles (this is what Ruben refers to on [one of his slides](https://rubenverborgh.github.io/SDSVoc-2016/#server-profile). We could just list all the profiles (comma-separated?)
* The requested resource is an XML document where the elements from namespace1 must conform to xml-schema1 and the elements from namespace2 must conform to xml-schema2. The current proposal only lists four URIs (which is the syntax used in xsi:schemaLocation). There is, however, currently no possibility to differentiate this from a simple list of profile-URIs all having q=1.0. Could we use a Map syntax here: {(namespace1,xml-schema1),(namespace2,xml-schema2)}?

**Comment by RubenVerborgh** [comment]
Case 1 is too limiting for RDF.

Case 2, I still support.

Case 3 seems too complex to me. However, different _fragments_ of a document could adhere to different syntaxes, as the `Link` header could have different fragments as source.

**Comment by larsgsvensson** [comment]
The point with case 3 is that it's a real one. When I showed an early version of the I-D to one of my colleagues (who probably is the one who will write the code to support it at our place), his first comment was about the use of multiple XML schemata,..

**Comment by RubenVerborgh** [comment]
I see. I'm still afraid about complexity; it would make this spec a) significantly more difficult b) specific to XML for a certain part.

**Comment by larsgsvensson** [comment]
Yep. But I'm afraid there is no one-size-fits-all here and that we might have to squeeze in some syntactic options that only apply to one (or two) media type(s)

**Comment by RubenVerborgh** [comment]
True; the question is just whether we want to go there. No, as far as I'm concerned, given that it makes client/server implementations more complex and hence less likely to happen. We could somehow try to make it optional (but then there's that problem).

**Comment by larsgsvensson** [comment]
If I remember your paper for Amsterdam correctly, you spoke about having documents that conform to combinations of profiles, too, So I think we cannot avoid looking at that problem anyway.

**Comment by RubenVerborgh** [comment]
Combinations of profiles can be as easy as just adding multiple profile links / attributes.

This discussion mostly reminds me that we'll need a requirements engineering document somewhere. Perhaps a new file in the repo, or a wiki page?

**Comment by larsgsvensson** [comment]
> This discussion mostly reminds me that we'll need a requirements engineering document somewhere. Perhaps a new file in the repo, or a wiki page?

Yeah, that would help. A wiki page sounds good.

**Comment by RubenVerborgh** [comment]
Tracking requirements in #16.
